packages/utils-debugging-protocol-common/src/debugging-protocol-connector.ts

```diff
@@ -22,26 +22,257 @@ import { Crdp } from 'chrome-remote-debug-protocol';
+    /** The payload returned by the debugging protocol in the `LoadingFinished` event. */
+    public get loadingFinished() {
+        return this._loadingFinished;
+    }
+
+    private _loadingFailed: Crdp.Network.LoadingFailedEvent | undefined;
+    /** The payload returned by the debugging protocol in the `LoadingFailed` event. */
+    public get loadingFailed() {
+        return this._loadingFailed;
+    }
+
+    private _responseBody: Crdp.Network.GetResponseBodyResponse | undefined;
+    public get responseBody() {
+        return this._responseBody;
+    }
+
+    private fetchContent(href: string, headers: HttpHeaders) {
+        const options = {
+            headers,
+            // we sync the ignore SSL error options with `request`. This is neeeded for local https tests
+            rejectUnauthorized: !overrideInvalidCert,
+            strictSSL: !overrideInvalidCert
+        };
+
+        const request: Requester = new Requester(options);
+
+        return request.get(href);
+    }
+
+    private _rawResponse: Buffer | undefined;
+    private getRawResponse() {
+        const that = this;
+
+        if (this._rawResponse) {
+            return Promise.resolve(this._rawResponse);
+        }
+
+        const { rawContent } = this._response!.body;
+
+        if (rawContent && rawContent.length.toString() === this._responseReceived!.response.headers['Content-Length']) {
+            // Response wasn't compressed so both buffers are the same
+            return Promise.resolve(rawContent);
+        }
+
+        const { url: responseUrl, requestHeaders: headers } = this._responseReceived!.response;
+
+        /*
+         * Real browser connectors automatically request using HTTP2. This spec has
+         * [`Pseudo-Header Fields`](https://tools.ietf.org/html/rfc7540#section-8.1.2.3):
+         * `:authority`, `:method`, `:path` and `:scheme`.
+         *
+         * An example of request with those `Pseudo-Header Fields` to google.com:
+         *
+         * ```
+         * :authority:www.google.com
+         * :method:GET
+         * :path:/images/branding/googlelogo/2x/googlelogo_color_120x44dp.png
+         * :scheme:https
+         * accept:text/html,application/xhtml+xml,application/xml;q=0.9,image/webp,image/apng,*\/*;q=0.8
+         * accept-encoding:gzip, deflate, br
+         * ...
+         * ```
+         *
+         * The `request` module doesn't support HTTP2 yet: https://github.com/request/request/issues/2033
+         * so the request need to be transformed to valid HTTP 1.1 ones, basically removing those headers.
+         *
+         */
+
+        const validHeaders = Object.entries(headers || {}).reduce((final, [key, value]) => {
+            if (key.startsWith(':')) {
+                return final;
+            }
+
+            final[key] = value;
+
+            return final;
+        }, {} as Crdp.Network.Headers);
+
+        return this.fetchContent(responseUrl, validHeaders)
```

Comment:
any reason to not use `async/await`?

Reply:
so the `catch` can be handle up by whoever is calling this method. This way I don't have to put a `try/catch` here.

Reply:
ok!